### DISCUSSION TITLE
[Feature Request] Stream Monitor - various additions

### DISCUSSION THREAD

**Discussion by superm1991** [ChristopherKlay/StadiaEnhanced#196] [Ideas & Requests]
Hi,

I think it would be useful to have a color indicator for packet loss :)
![image](https://user-images.githubusercontent.com/4275183/142721131-42117a46-7d04-4be4-bc7c-fc2cc0c90e03.png)

Also: I don't know if this already exists, but it'd be nice to have a way to display only the 4 color indicators instead of the whole Stream Monitor

And maybe a little button to reset the counters, if things drastically improve/degrade in the middle of a long session?

Okay... I'll stop with my christmas list lol

Thanks again!

**Comment by axelkunz** [comment]
You can double-click on the streaming monitor to switch to a more compact UI. It has a single color indicator that is a combination of the colors in the full UI.

**Reply by superm1991** [reply]
Thanks for the info :)
Does it take packet loss into account? because there's still no color indicator for packet loss... and that usually the main issue with my connection

**Reply by axelkunz** [reply]
Currently not. Framedrop%, latency and decoding time is used for the calculation of the color. 

Adding a color indicator for packet loss would be possible in theory, but I am not sure if we are able to define reasonable threshold values for each color. @ChristopherKlay what do you think?

**Reply by ChristopherKlay** [reply]
I think the main issue is that packet loss isn't a "X% means you will have issues" deal and indicating it like that wouldn't really make sense.

In pretty much every case of packet loss that would affect your experience, you would be dropping frames due to it to begin with.

**Reply by superm1991** [reply]
I can **definitely** feel packet loss affecting gameplay on a driving game like The Crew 2, even without frame drops (screenshot above). The car  not turning when you press the button is a dead giveaway... (I also remember micro-freezes in the audio)

I don't know about network and tech, but maybe in my case it was **upload** packet loss as well alongside regular download packet loss? And that's why it affected gameplay as much as it did?

Regarding the threshold:
- approaching **0.5%** I'd say it starts to affect gameplay, though you might not notice _immediately_
- above **1%** (screenshot above) gameplay is seriously degraded

**Comment by ChristopherKlay** [comment]
The issue is, 0.5% doesn't **always** affect gameplay, because it depends purely on what packets are actually lost, unlike i.e. frame drops, high latency and other factors, that always do.

It also depends on the used video codec; VP9 suffers significantly more from lost packets compared to H264 from my experience, because the two codecs act different when performance issues are present. Instead of dropping inputs and stuttering, H264 simply reduces the quality but stays fairly stable in terms of inputs.

The issue is also things like; How do we detect what "perfect" means in terms of packet loss? It can't be 0%, because a lot of people do randomly drop a couple of frames/packets on game start and thus wouldn't ever reach 0% - if we make it 0.1% however, someone *could* definitely still feel those packets drop in a worst-case scenario, especially if it's just spikes every X minutes.

**Reply by superm1991** [reply]
Oh okay! I see what you mean.

Let's forget about the color indicator for packet loss.
And @SHANYUAN already gave me an answer for the second part of the request.
But maybe the third part of the request could still be a good idea, especially with your comment about spikes?
> And maybe a little button to reset the counters, if things drastically improve/degrade in the middle of a long session?

**Reply by ChristopherKlay** [reply]
I believe there's a even better solution for the entire "stats are averaged out in longer sessions" issue on the way already. (:
Will definitely keep that in mind!

**Reply by superm1991** [reply]
Awesome! Thanks a lot :)
